src/main/java/seedu/address/logic/candidate/EditCandidateCommand.java

```diff
@@ -105,12 +105,29 @@ public CommandResult execute(Model model) throws CommandException {
         }
         editedPerson.setPositions(positionReferences);
 
-        //Remove the old person and add the new one
         Set<Interview> interviews = personToEdit.getInterviews();
-        for (Interview i : interviews) {
-            i.deleteCandidate(personToEdit);
-            i.addCandidate(editedPerson);
+
```

Comment:
You need to add a line `newPositions = positionReferences;`. This is to fix the bug when the user does not input the exact same casing as the position in the position list. For example, edit_c 2 position=BOOKKEEPER still unassigns the Bookkeeper interview from a candidate

Reply:
Ah thanks for catching this. Resolved.